utilities/db-migration/migrations/000013_v13.up.sql

```diff
@@ -0,0 +1,44 @@
+-- AppProjectRepository is used by ArgoCD AppProject
+CREATE TABLE AppProjectRepository (
+
+    -- Primary Key, that is an auto-generated UID
+	app_project_repository_id VARCHAR(48) NOT NULL PRIMARY KEY,
+
+	-- Describes whose cluster this is (UID)
+	-- Foreign key to: ClusterUser.clusteruser_id
+	clusteruser_id VARCHAR (48),
+	CONSTRAINT fk_clusteruser_id FOREIGN KEY (clusteruser_id) REFERENCES ClusterUser(clusteruser_id) ON DELETE NO ACTION ON UPDATE NO ACTION,
+
+	-- Describes which repositorycredentials the user has access to (UID)
+	-- Foreign key to: RepositoryCredentials.repositorycredentials_id
+	repositorycredentials_id VARCHAR ( 48 ),
+	CONSTRAINT fk_repositorycredentials_id FOREIGN KEY (repositorycredentials_id) REFERENCES RepositoryCredentials(repositorycredentials_id) ON DELETE SET NULL ON UPDATE NO ACTION,
+	
+	-- Normalized Repo URL
+	repo_url VARCHAR (256) NOT NULL,
+
+	UNIQUE (clusteruser_id, repo_url),
+
+	seq_id serial
+);
+-- Add an index on clusteruser_id
+CREATE INDEX idx_userid_cluster_rc ON AppProjectRepository(clusteruser_id);
+
+-- AppProjectManagedEnvironment is used by ArgoCD AppProject
+CREATE TABLE AppProjectManagedEnvironment (
+
+    -- Primary Key, that is an auto-generated UID
+	app_project_managedenv_id VARCHAR(48) NOT NULL PRIMARY KEY,
+
+	-- Describes whose cluster this is (UID)
+	clusteruser_id VARCHAR (48),
```

Comment:
As mentioned above, it was an oversight on my part: clusteruser_id should definitely be a foreign key to clusteruser table, and it should also be NOT NULL